src/block/Anvil.php

```diff
@@ -95,4 +98,15 @@ public function place(BlockTransaction $tx, Item $item, Block $blockReplace, Blo
 		}
 		return parent::place($tx, $item, $blockReplace, $blockClicked, $face, $clickVector, $player);
 	}
+
+	public function onHitGround(FallingBlock $blockEntity) : ?Block{
+		if(lcg_value() < 0.05 + (round($blockEntity->getFallDistance()) - 1) * 0.05){
+			if($this->damage !== self::VERY_DAMAGED){
+				$this->damage = $this->damage + 1;
+			}else{
+				return null;
+			}
+		}
+		return $this;
```

Comment:
Is there any particular reason to return a Block here? It seems to me that a boolean would be sufficient.

Reply:
`Returns the block resulting from hitting the ground, null will cause no block to be placed`
Allow to change the resulting block, not necessary for this implementation but likely in later implementations.

Reply:
Such as?

Reply:
hmm, no block in vanilla, I will do your suggestion.